net-misc/curl/curl-7.48.0.recipe

```diff
@@ -1,16 +1,18 @@
 SUMMARY="A commandline-tool and library for downloading data from URLs"
-DESCRIPTION="Curl is a command line tool for transferring data with URL \
+DESCRIPTION="curl is a command line tool for transferring data with URL \
 syntax, supporting DICT, FILE, FTP, FTPS, Gopher, HTTP, HTTPS, IMAP, IMAPS, \
-LDAP, LDAPS, POP3, POP3S, RTMP, RTSP, SCP, SFTP, SMTP, SMTPS, Telnet and TFTP. curl supports SSL certificates, HTTP POST, HTTP PUT, FTP uploading, HTTP form \
+LDAP, LDAPS, POP3, POP3S, RTMP, RTSP, SCP, SFTP, SMTP, SMTPS, Telnet and TFTP.
+curl supports SSL certificates, HTTP POST, HTTP PUT, FTP uploading, HTTP form \
 based upload, proxies, cookies, user+password authentication (Basic, Digest, \
 NTLM, Negotiate, kerberos...), file transfer resume, proxy tunneling and a \
 busload of other useful tricks."
-HOMEPAGE="http://curl.haxx.se"
+HOMEPAGE="https://curl.haxx.se/"
 COPYRIGHT="1996-2016, Daniel Stenberg"
 LICENSE="Curl"
 REVISION="1"
-SOURCE_URI="http://curl.haxx.se/download/curl-$portVersion.tar.bz2"
-CHECKSUM_SHA256="ddc643ab9382e24bbe4747d43df189a0a6ce38fcb33df041b9cb0b3cd47ae98f"
+SOURCE_URI="https://curl.haxx.se/download/curl-$portVersion.tar.bz2"
+#CHECKSUM_SHA256="ddc643ab9382e24bbe4747d43df189a0a6ce38fcb33df041b9cb0b3cd47ae98f"
```

Comment:
There are a few problems with the current layout because there are a lot of html files in **`develop/documentation/curl/`** and they are both in curl_devel and curl_x86_devel.

The split layout I'm proposing is described in the following table, where:
* 1=primary arch
* 2=secondary arch
* 1dev=primary arch devel
* 2dev=secondary arch devel

pkg | files | moved to
--- | ----- | -----------
1 |	bin/curl
1 |	data/aclocal/libcurl.m4			| common
1 |	data/licenses/Curl			| common
1 |	documentation/man/man1/curl-config.1	| common
1 |	documentation/man/man1/curl.1		| common
1 |	documentation/packages/curl/curl.html	| common
1 |	lib/libcurl.so.4
1 |	lib/libcurl.so.4.4.0
||
2 |	data/aclocal/libcurl.m4		| common
2 |	data/licenses/Curl		| common
2 |	lib/x86/libcurl.so.4
2 |	lib/x86/libcurl.so.4.4.0
||
1dev |	bin/curl-config
1dev |	data/licenses/Curl			| common
1dev |	develop/documentation/curl/*.html	| common_devel
1dev |	develop/headers/curl/*.h
1dev |	develop/lib/libcurl.*
1dev |	develop/lib/pkgconfig/libcurl.pc
1dev |	documentation/man/man3/*.3		| common_devel
||
2dev |	bin/curl-config-x86
2dev |	bin/x86/curl-config
2dev |	data/licenses/Curl			| common
2dev |	develop/documentation/curl_x86/*.html	| common_devel
2dev |	develop/headers/x86/curl/*.h
2dev |	develop/lib/x86/libcurl.*
2dev |	develop/lib/x86/pkgconfig/libcurl.pc
2dev |	documentation/man/man3/*.3		| common_devel

With this layout we save:
* about 286 kiB which would otherwise be duplicated in curl_devel and curl_x86_devel;
* about 84 kiB which would otherwise be duplicated in curl and curl_x86.

Moreover, this layout allows to install only what you need. If you don't want to waste 270kiB with curl_devel  but still be able to install curl_x86_devel, you can do that.

Here are the sizes for curl-7.47.1 with the new layout:
```
-rw-r--r-- 1 305159 Mar 23 19:02 curl-7.47.1-1-x86_gcc2.hpkg
-rw-r--r-- 1  83885 Mar 23 19:02 curl_common-7.47.1-1-any.hpkg
-rw-r--r-- 1 290785 Mar 23 19:02 curl_common_devel-7.47.1-1-any.hpkg
-rw-r--r-- 1 274328 Mar 23 19:02 curl_devel-7.47.1-1-x86_gcc2.hpkg
-rw-r--r-- 1 204318 Mar 23 19:30 curl_x86-7.47.1-1-x86_gcc2.hpkg
-rw-r--r-- 1 308317 Mar 23 19:30 curl_x86_devel-7.47.1-1-x86_gcc2.hpkg
```

If we keep the old layout, then we should at least remove the duplicate files. But there is a consequence: curl_x86_devel will need to depend on curl_devel, otherwise you need to ship all html files in each devel package.

If you prefer the old layout, then I can switch back, but I think it would be much better with the new layout because we have real architecture-independent packages, we avoid package overlaps, and we save space.